force-app/main/default/classes/LeadTriggerHandlerTest.cls

```diff
@@ -0,0 +1,128 @@
+ * the org size limit for all Apex scripts.
+ *
+ * See the Apex Language Reference for more information about Testing and Code Coverage.
+ */
+@isTest
+private class LeadTriggerHandlerTest {
+
+    @testSetup
+    static void setupTestData() {
+        List<Lead> leads = new List<Lead>{
+            new Lead(FirstName='John', LastName='Doe', Company='UniqueTech' + System.currentTimeMillis(), Title='vp', Email='test' + System.currentTimeMillis() + '@example.com', Phone='1234567890', Industry='Technology', LeadSource='Website'),
+            new Lead(FirstName='Jane', LastName='Smith', Company='UniqueBiz' + System.currentTimeMillis(), Title='mgr', Email='test2' + System.currentTimeMillis() + '@example.com', Phone='9876543210', Industry='Retail', LeadSource='Referral'),
+            new Lead(FirstName='Test', LastName='Coverage', Company='UniqueBizzzzz' + System.currentTimeMillis(), Title='exec', Email='test2' + System.currentTimeMillis() + '@example2.com', Phone='9876543266', Industry='Retailer', LeadSource='Referralll'),
+            new Lead(FirstName='Testing', LastName='Coveraged', Company='UniqueBizzzxz' + System.currentTimeMillis(), Title='assist', Email='test21' + System.currentTimeMillis() + '@example1.com', Phone='9878543266', Industry='Retasiler', LeadSource='Refesrralll')
+
+
+        };
+        insert leads;
+
+        Contact contact = new Contact(FirstName='John', LastName='Doe', Email='unique_contact' + System.currentTimeMillis() + '@example.com');
+        insert contact;
+    }
+
+
+
+  
+    
+    @isTest
+    static void testHandleTitleNormalization() {
+        List<Lead> leads = [SELECT Id, Title FROM Lead];
+        Test.startTest();
+        LeadTriggerHandler.handleTitleNormalization(leads);
+        Test.stopTest();
+
+        for (Lead lead : leads) {
+            if (lead.Title == 'vp') {
+                System.assertEquals('Vice President', lead.Title, 'Title should be normalized to Vice President');
+            } else if (lead.Title.contains('mgr')) {
+                System.assertEquals('Manager', lead.Title, 'Title should be normalized to Manager');
+            } else if (lead.Title.contains('exec')) {
+                System.assertEquals('Executive', lead.Title, 'Title should be normalized to Executive');
+            } else if (lead.Title.contains('assist')) {
+                System.assertEquals('Assistant', lead.Title, 'Title should be normalized to Assistant');
+            }
+        }
+    }
+
+    @isTest
+    static void testHandleAutoLeadScoring() {
+        List<Lead> leads = [SELECT Id, LeadSource, Email, Phone, Industry, Lead_Score__c FROM Lead];
+        Test.startTest();
+        LeadTriggerHandler.handleAutoLeadScoring(leads);
+        Test.stopTest();
+
+        for (Lead lead : leads) {
+            if (lead.LeadSource == 'Website' && lead.Email != null) {
+                System.assert(lead.Lead_Score__c >= 3, 'Score should account for website source and email presence');
```

Comment:
```suggestion
            if (lead.LeadSource == 'Website' && lead.Email != null) {
                System.assert(lead.Lead_Score__c = 3, 'Score should should be exactly 3');
```
This seems to be a little more accurate, You wouldn't want to leave room later for a lead score of 1 or 2 to get through and pass the test.
Same for the below lines